File: php/features.php
<?php
declare(strict_types=1);

// Geonode SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GeonodeFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GeonodeBaseFeature();
            case "test":
                return new GeonodeTestFeature();
            default:
                return new GeonodeBaseFeature();
        }
    }
}

File: py/features.py
# Geonode SDK feature factory

from feature.base_feature import GeonodeBaseFeature
from feature.test_feature import GeonodeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GeonodeBaseFeature(),
        "test": lambda: GeonodeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
